Source/WebCore/platform/gstreamer/WebKitFliteSourceGStreamer.cpp

```diff
@@ -0,0 +1,263 @@
+/*
+ *  Copyright (C) 2023 Igalia, S.L
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include "config.h"
+#include "WebKitFliteSourceGStreamer.h"
+
+#if ENABLE(SPEECH_SYNTHESIS) && USE(GSTREAMER)
+
+#include "GStreamerCommon.h"
+#include "PlatformSpeechSynthesisVoice.h"
+#include <flite/flite.h>
+#include <wtf/DataMutex.h>
+#include <wtf/glib/WTFGType.h>
+
+extern "C" {
+/* There is no header for Flite functions below */
+void usenglish_init(cst_voice*);
+cst_lexicon* cmulex_init(void);
+cst_voice* register_cmu_us_kal(const char*);
+cst_voice* register_cmu_us_awb(const char*);
+cst_voice* register_cmu_us_rms(const char*);
+cst_voice* register_cmu_us_slt(const char*);
+}
+
+using namespace WebCore;
+
+typedef struct _WebKitFliteSrcClass WebKitFliteSrcClass;
+typedef struct _WebKitFliteSrcPrivate WebKitFliteSrcPrivate;
+
+#define WEBKIT_FLITE_SRC_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST((klass), WEBKIT_TYPE_FLITE_SRC, WebKitFliteSrcClass))
+#define WEBKIT_IS_FLITE_SRC(obj) (G_TYPE_CHECK_INSTANCE_TYPE((obj), WEBKIT_TYPE_FLITE_SRC))
+#define WEBKIT_IS_FLITE_SRC_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE((klass), WEBKIT_TYPE_FLITE_SRC))
+
+struct _WebKitFliteSrcClass {
+    GstBaseSrcClass parentClass;
+};
+
+struct _WebKitFliteSrc {
+    GstBaseSrc parent;
+    WebKitFliteSrcPrivate* priv;
+};
+
+struct _WebKitFliteSrcPrivate {
+    struct StreamingMembers {
+        bool didLoadUtterance;
+        GRefPtr<GstAdapter> adapter;
+    };
+    DataMutex<StreamingMembers> dataMutex;
+
+    GstAudioInfo info;
+    String text;
+    cst_voice* voice;
```

Comment:
I suspect this should be stored in a GUniquePtr calling `delete_voice()` to release resources?

Reply:
Done. I applied  GUniquePtr to cst_voice*.